packages/app/src/cli/services/dev.ts

```diff
@@ -159,6 +164,7 @@ async function prepareForDev(commandOptions: DevOptions): Promise<DevConfig> {
     partnerUrlsUpdated,
     graphiqlPort,
     graphiqlKey,
+    consistentDev: developerPlatformClient.clientName === ClientName.AppManagement,
```

Comment:
Instead of checking the name, we could add a new property like these ones: https://github.com/Shopify/cli/blob/866af380a46018ed1168a0ef15e4cfa36d24f165/packages/app/src/cli/utilities/developer-platform-client/app-management-client.ts#L158-L159